src/js/yeldaWebchatInjector.js

```diff
@@ -949,6 +969,55 @@ class YeldaChat {
       webchatFrame.contentWindow.postMessage({ event: 'sendUserMessage', data: message })
     }
   }
+
+  /**
+   * Listen leave viewport event handling
+   * @param {Boolean} shouldRemoveEvent - default false
+   */
+  listenLeaveViewport(shouldRemoveEvent = false) {
+    const md = new MobileDetect(navigator.userAgent)
+    const isMobile = md.mobile() !== null
+
+    /**
+     * Not needed to listen on mobile since webchat covers the full page of the site
+     * and we can't able to detect mouseout events
+     */
+    if (isMobile) {
+      return
+    }
+
+    const eventHandler = window[shouldRemoveEvent ? 'removeEventListener' : 'addEventListener']
+
+    if (!this.viewportListenerBind) {
+      this.viewportListenerBind = this.viewportListener.bind(this)
+    }
+    // mouseout event is used to listen leave viewport
+    eventHandler('mouseout', this.viewportListenerBind)
+
+    if (shouldRemoveEvent) {
+      this.pushNotificationId = null
+    }
+  }
+
+  /**
+   * Callback for leave viewport listener
+   * @param {Event} event
+   */
+  viewportListener(event) {
+    /**
+     * when mouse moves between dom elements mouseout event is trigged but we need to trigger
+     * the event when mouse moves out of the viewport so toElement and relatedTarget is checked
+     */
+    if (event.toElement || event.relatedTarget) {
+      return
+    }
+
+    // When mouse leaves the viewport send 'leaveViewPort' to webchat with the notificationId
+    document.getElementById('web_chat_frame').contentWindow.postMessage({
+      event: 'leaveViewPort',
+      pushNotificationId: this.pushNotificationId
+    }, '*')
```

Comment:
why do we need to send the pushNotificationId ? can't the webchat handle by itself the event and find the push notification id to send by itself ?